src/analysis/special_functions/pow.lean

```diff
@@ -880,7 +885,22 @@ open filter
 lemma filter.tendsto.nnrpow {α : Type*} {f : filter α} {u : α → ℝ≥0} {v : α → ℝ} {x : ℝ≥0} {y : ℝ}
   (hx : tendsto u f (𝓝 x)) (hy : tendsto v f (𝓝 y)) (h : x ≠ 0 ∨ 0 < y) :
   tendsto (λ a, (u a) ^ (v a)) f (𝓝 (x ^ y)) :=
-tendsto.comp (nnreal.continuous_at_rpow h) (tendsto.prod_mk_nhds hx hy)
+tendsto.comp (nnreal.continuous_at_rpow h) (hx.prod_mk_nhds hy)
+
+namespace nnreal
+
+lemma continuous_at_rpow_const {x : ℝ≥0} {y : ℝ} (h : x ≠ 0 ∨ 0 ≤ y) :
```

Comment:
I find this formulation (with the ` ∨` hypothesis) a little odd, is there a reason not to have two separate lemmas?

Reply:
We can have three lemmas. I like the idea of having the most powerful lemma with a predictable name because this way I can `apply nnreal.continuous_at_rpow` or `apply nnreal.continuous_at_rpow_const`, then look at the new goals.

Another approach is to actually prove an `iff` lemma, and make it `@[simp]`.

Reply:
> I like the idea of having the most powerful lemma with a predictable name because this way I can `apply nnreal.continuous_at_rpow` or `apply nnreal.continuous_at_rpow_const`, then look at the new goals.

OK, makes sense!